src/components/Booking/WarningMessage/index.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { View } from 'react-native';
+import { PropTypes as PT } from 'prop-types';
+import { P } from '../../Common';
+import styles from '../View/styles';
+import { RED } from '../../../styles/colors';
+
+const WarningMessage = (props) => {
+  const {
+    remainingHolidays,
+    booked,
+    potentialHolidays,
+    booking,
+  } = props;
+
+  const gettingMessage = getMessage(
+    remainingHolidays,
+    booked,
+    potentialHolidays,
+    booking
+  );
+
+  return (
+    (gettingMessage !== '')
+      ? (
+        <View style={styles.warningMessage}>
+          <P style={{ color: RED }}>
+            {gettingMessage}
+          </P>
+        </View>)
+      : null
+  );
+};
+
+const getMessage = (
+  remainingHolidays,
+  booked,
+  potentialHolidays,
+  booking,
+) => {
+  const { duration = 0 } = booking;
+
+  const remainingHolidaysForNewBookings = booked
+    ? ((remainingHolidays - potentialHolidays + duration).toFixed(1))
+    : ((remainingHolidays - potentialHolidays).toFixed(1));
+
+  let message = '';
+  const bookingWarning = 'You are unable to book this holiday due to not having enough available hoildays. Please contact HR.';
+
+  if ((booked && remainingHolidaysForNewBookings < 0)
+    || (!booked && remainingHolidaysForNewBookings < 0)) {
```

Comment:
i think you just need `if(remainingHolidaysForNewBookings < 0) {` here instead of if `booked = true` OR `booked = false`

Reply:
ah yes. i have changed same

Reply:
i have also made use of the else if